Shopify/cli
# Add abortController to async ui function parameters

### WHY are these changes introduced?

At the moment there is no way for async processes to make the async prompts quit. For example in this scenario, the prompt would still remain in the terminal:

```
const templatesPromise = getLatestTemplates()
    .catch((error) => {
      renderFatalError(error);
      process.exit(1);
    });

await renderXPrompt(...);
```

![image (6)](https://user-images.githubusercontent.com/151725/234042022-868dfabc-8bc4-4913-be10-7fc4fb19ef6c.png)

### WHAT is this pull request doing?

This PR adds the ability to pass an abort controller from outside. Prompts and `Tasks` will clear the output before quitting when the abort controller signal receives the `abort` event.
`ConcurrentOutput` will clear only the dynamic part, currently the preview URL and the shortcuts, but will preserve the logs.

### How to test your changes?

I've tested it modifying `kitchen-sink prompts` and replacing the first prompt with this code

```
  const controller = new AbortController()

  // after 2 seconds call abort
  setTimeout(() => {
    controller.abort()
  }, 2000)

  // renderSelectPrompt
  await renderSelectPrompt({
    message: 'Associate your project with the org Castile Ventures?',
    choices: [
      {label: 'first', value: 'first', key: 'f'},
      {label: 'second', value: 'second', key: 's'},
      {label: 'third', value: 'third'},
      {label: 'fourth', value: 'fourth'},
      {label: 'fifth', value: 'fifth', group: 'Automations', key: 'a'},
      {label: 'sixth', value: 'sixth', group: 'Automations'},
      {label: 'seventh', value: 'seventh'},
      {label: 'eighth', value: 'eighth', group: 'Merchant Admin'},
      {label: 'ninth', value: 'ninth', group: 'Merchant Admin'},
      {label: 'tenth', value: 'tenth'},
    ],
    infoTable: {add: ['new-ext'], remove: ['integrated-demand-ext', 'order-discount']},
    abortController: controller,
  })
```

The prompt should disappear.
